tutorial/chapter1.md

```diff
@@ -1,41 +1,219 @@
+```
+
+Typical output looks like:
+
+```text
+[2023-09-04 13:59:47.280] [info] Read a list of 5 images in folder: tests/data
+[2023-09-04 13:59:47.280] [info] Read a list of 5 images in folder: tests/data
+[2023-09-04 13:59:47.280] [info] Read a list of 5 images in folder: tests/data
+[2023-09-04 13:59:47.280] [info] Service Runner 0. Iteration 0. Thread-id: 139991226636032
+[2023-09-04 13:59:47.280] [info] Service Runner 1. Iteration 0. Thread-id: 139991218243328
+[2023-09-04 13:59:47.280] [info] Service Runner 2. Iteration 0. Thread-id: 139991209850624
+[2023-09-04 13:59:47.466] [info] Service Runner 1. Iteration 1. Thread-id: 139991218243328
+[2023-09-04 13:59:47.469] [info] Listener 1. Thread-id: 139991310563072
+[2023-09-04 13:59:47.514] [info] Service Runner 2. Iteration 1. Thread-id: 139991209850624
+[2023-09-04 13:59:47.517] [info] Listener 2. Thread-id: 139991310563072
+[2023-09-04 13:59:47.528] [info] Listener 0. Thread-id: 139991310563072
+[...]
+[2023-09-04 13:59:48.406] [info] Service Runner 1. Iteration 5. Thread-id: 139991218243328
+[2023-09-04 13:59:48.406] [info] Read a list of 5 images in folder: tests/data
+[2023-09-04 13:59:48.422] [info] Listener 2. Thread-id: 139991310563072
+[2023-09-04 13:59:48.496] [info] Listener 0. Thread-id: 139991310563072
+[2023-09-04 13:59:48.600] [info] Listener 1. Thread-id: 139991310563072
+[2023-09-04 13:59:48.622] [info] Total frames listened: 18.
+```
+
+## Event loop example 2
+
+This example is similar to example 1: it simulates 2 cameras' images process by resizing and converting to B/W. See the [source code](../examples/event_loop/src/event_loop_example2.cpp) and run:
+
+```bash
+build/bin/kpsr_image_processing_eventloop_example_2 <input_images_folder> <output_images_folder> [debug]
+```
+
+The code includes additional improvements:
+
+- Each camera has its own even loop (and publisher).
+    Note that every event loop could be assigned to different CPUs and so reserve more resources for the most demanding task.
+- One additional event loop for the images' transformation.
+- Each camera has two subscribers: one for the images reading and one for transformation, with the help of `kpsr::EventTransformForwarder`:
+
+```cpp
+std::vector<std::unique_ptr<
+    kpsr::EventTransformForwarder<kpsr::vision_ocv::ImageData, kpsr::vision_ocv::ImageData>>>
+    eventTransformers;
+[...]
+    eventTransformers.emplace_back(
+        new kpsr::EventTransformForwarder<kpsr::vision_ocv::ImageData,
+                                          kpsr::vision_ocv::ImageData>(
+            toGrayscaleFunction, publishersTransformedImages[i]))
+```
+
+see [Example 7: The EventTransformForwarder](https://github.com/klepsydra-technologies/kpsr-tutorial/blob/main/tutorials/chapter1.md#example-7-the-eventtransformforwarder) in the main Klepsydra tutorial for additional information and advantages: simpler event creation and publishing, and cleaner code more akin to the map pattern.
+
+Typical output looks like:
```

Comment:
```suggestion
In contrast with the previous example, in the next output, listeners of`ToGrayScaleConverter` do not share the same thread id because they are running on different *EventLoops*. All listeners of the writer share it because they belong to the same *EventLoop*.
```